src/formState.test.tsx

```diff
@@ -1552,6 +1552,55 @@ describe("formState", () => {
     // The list is again dirty
     expect(a1.books.dirty).toBeTruthy();
   });
+
+  describe("fragments", () => {
+    type BookWithFragment = BookInput & { data: Fragment<{ foo: string }> };
+
+    it("ignores changes within fragments", () => {
+      const a: ObjectState<BookWithFragment> = createObjectState<BookWithFragment>(
+        { title: { type: "value" }, data: { type: "fragment" } },
+        { title: "b1", data: fragment({ foo: "1" }) },
+      );
+      let numCalcs = 0;
+      autorun(() => {
+        numCalcs++;
+        noop(a.value);
+      });
+      expect(a.data.value).toEqual({ foo: "1" });
+      expect(isObservable(a.data.value)).toBe(false);
+      expect(numCalcs).toEqual(1);
+      a.data.value = { foo: "2" };
+      expect(numCalcs).toEqual(2);
+      a.data.value.foo = "3";
```

Comment:
I'm wondering if I'm interpreting this correctly.  If we change the entire fragment object, that change is recognized in the autorun. However, if we try to change a single value within the fragment itself, the change is ignored?

Reply:
@pwillikins Yep!

The idea is that `data` might be a ~huge JSON fragment, so we purposefully don't want to watch/react to the whole thing.

(That is historically what has kept us from "just putting the whole fragment in the form" and instead doing our "choose ~3-4-5-wait-now-10 values out of the fragment that we _really need_ and store them on the form. That said, we were totally doing this as premature optimization b/c mobx in general is pretty fast and honestly it probably would have been fine--to point the whole fragment directly in the form state _without_ this PR's special `fragment` support. But :shrug: now with the official `fragment` support it should be definitely okay.)

_But_ maybe after a save/something, we get back the latest top-level fragment, and so sure switch out `authorFragment1` --> `authorFragment2` and any components that had been using `authorFragment1` via `form.data.value` will re-render.